libraries/chain/include/scorum/chain/services/proposal_executor.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <scorum/chain/services/dbs_base.hpp>
+#include <scorum/chain/evaluators/evaluator_registry.hpp>
+#include <scorum/protocol/proposal_operations.hpp>
+
+namespace scorum {
+namespace chain {
+
+struct data_service_factory_i;
+struct proposal_service_i;
+class proposal_object;
+
+struct proposal_executor_service_i
+{
+    virtual void operator()(const proposal_object& proposal) = 0;
+};
+
+class dbs_proposal_executor : public dbs_base, public proposal_executor_service_i
```

Comment:
Bad solution to use dbs_base for this class. dbs-service intend to cover only db-index operations.
This class should use new factory to support interface proposal_executor_service_i. New type of objects.